packages/ui/src/component/UIGroup.ts

```diff
@@ -0,0 +1,226 @@
+import {
+  Component,
+  ComponentType,
+  DisorderedArray,
+  Entity,
+  EntityModifyFlags,
+  assignmentClone,
+  ignoreClone
+} from "@galacean/engine";
+import { Utils } from "../Utils";
+import { IGroupAble } from "../interface/IGroupAble";
+import { EntityUIModifyFlags, UICanvas } from "./UICanvas";
+
+export class UIGroup extends Component implements IGroupAble {
+  /** @internal */
+  @ignoreClone
+  _indexInGroup: number = -1;
+  /** @internal */
+  @ignoreClone
+  _indexInCanvas: number = -1;
+  /** @internal */
+  @ignoreClone
+  _group: UIGroup;
+  /** @internal */
+  @ignoreClone
+  _canvas: UICanvas;
+  /** @internal */
+  @ignoreClone
+  _disorderedElements: DisorderedArray<IGroupAble> = new DisorderedArray();
+
+  /** @internal */
+  @ignoreClone
+  _globalAlpha = 1;
+  /** @internal */
+  _globalInteractive = true;
+
+  @assignmentClone
+  private _alpha = 1;
+  @assignmentClone
+  private _interactive = true;
+  @assignmentClone
+  private _ignoreParentGroup = false;
+  @ignoreClone
+  private _entityListeners: Entity[] = [];
+
+  /** @internal */
+  @ignoreClone
+  _canvasListeningEntities: Entity[] = [];
+  /** @internal */
+  @ignoreClone
+  _groupListeningEntities: Entity[] = [];
+  /** @internal */
+  @ignoreClone
+  _isCanvasDirty: boolean = true;
+  /** @internal */
+  @ignoreClone
+  _isGroupDirty: boolean = true;
+  /** @internal */
+  @ignoreClone
+  _groupDirtyFlags: number = GroupModifyFlags.None;
+
+  /**
+   * @internal
+   */
+  get canvas(): UICanvas {
+    if (this._isCanvasDirty) {
+      const curCanvas = Utils.getCanvasInParents(this.entity);
+      Utils._registerElementToCanvas(this, this._canvas, curCanvas);
+      Utils._registerElementToCanvasListener(this, curCanvas);
+      this._isCanvasDirty = false;
+    }
+    return this._canvas;
+  }
+
+  /**
+   * @internal
+   */
+  get group(): UIGroup {
+    if (this._isGroupDirty) {
+      const canvas = this.canvas;
+      Utils._registerElementToGroup(this, this._group, Utils.getGroupInParents(this.entity, canvas?.entity));
+      Utils._registerElementToGroupListener(this, canvas);
+      this._isGroupDirty = false;
+    }
+    return this._group;
+  }
```

Comment:
If get canvas and get group  appear to have the same implementation in UIGroup/UIRenderer/UIInteractive, can they be considered for abstraction?